run.py

```diff
@@ -0,0 +1,123 @@
+import torch
+
+import spikeinterface as si
+import spikeinterface.sorters as ss
+
+import os
+import sys
+import zipfile
+import boto3
+
+from kilosort import io, run_kilosort
+from spikeinterface.extractors import read_nwb_recording, read_maxwell
+
+
+this_dir = os.path.dirname(__file__)
+output_dir = os.path.join(this_dir, 'output')
+
+
+def zip_files(dir: str, zip_filename: str):
+    with zipfile.ZipFile(zip_filename, 'w') as z:
+        for root, dirs, files in os.walk(dir, topdown=False):
+            for f in files:
+                f = os.path.join(root, f)
+                z.write(f, os.path.basename(f))
+
+
+def mock_run_sorter(input_filepath: str):
+    os.makedirs(output_dir, exist_ok=True)
+    for file in ['a.npy', 'b.npy', 'c.npy']:
+        with open(os.path.join(output_dir, file), 'w') as f:
+            f.write(f'{file} has content: {file}')
+    return input_filepath
+
+
+def main(mock: bool = False):
+    input_filepath: str = sys.argv[1]
+    s3_client = boto3.client('s3', endpoint_url='https://s3-west.nrp-nautilus.io')
+
+    if input_filepath.startswith('s3://'):
+        print(f'Downloading {input_filepath} with boto3 version: {boto3.__version__}')
+        bucket, key = input_filepath[len('s3://'):].split('/', maxsplit=1)
+        s3_client.download_file(bucket, key, f'/tmp/{os.path.basename(input_filepath)}')
+    else:
+        raise RuntimeError(f'Input filepath must start with s3:// !  {input_filepath}')
```

Comment:
Sometimes it's clearer to use the `if` solely to check for the exceptional condition and eject if necessary, and keep the "mainline" logic on the "mainline".  For example:
```
if not input_filepath.startswith('s3://'):
    raise RuntimeError(f'Input filepath must start with s3:// !  {input_filepath}')

print(f'Downloading {input_filepath} with boto3 version: {boto3.__version__}')
bucket, key = input_filepath[len('s3://'):].split('/', maxsplit=1)
s3_client.download_file(bucket, key, f'/tmp/{os.path.basename(input_filepath)}')
```
Of course, there's pros and cons to either.